src/index.ts

```diff
@@ -160,29 +164,57 @@ app.get('/browserAuthenticate.js', (req, res) => {
   res.sendFile(browserPath +'/dist/index.js');
 });
 
-app.get("/azureRedirect", async(req: any, res: any) => {
-  // The redirect will either contain a "code" or an "error"
+app.get("/startAzureRedirect",(req: express.Request,res: express.Response)=>{
+  const credential = new WebRedirectCredential( "27029f03-7c64-4ef6-88e4-14539e6c8d8c", 
+  "ed0e15fd-bb43-44e2-8d9c-b9c9758e0a46","http://localhost:8080/azureRedirect");
+  const state = (req.session as any).user;
+  const scope = "https://graph.microsoft.com/.default";
+  const authorizeUrl = credential.getAuthorizeUrl(scope,{state});
+  res.redirect(authorizeUrl);
+})
+
+app.get("/azureRedirect", async(req,res)=>{
   const authorizationCode = req.query["code"];
-  setLogLevel("verbose");
-  console.log(`authorization code = ${authorizationCode}`);
-  //for auth code cred, we need web for registering the redirect URIs
-  const authCred = new AuthorizationCodeCredential(
-    "27029f03-7c64-4ef6-88e4-14539e6c8d8c", 
-    "747a3a69-568f-4d40-9e9c-8f21472f246e",
-    authorizationCode,
```

Comment:
One advantage of AuthorizationCodeCredential is that the customers may find it easier to pass the authorizationCode on the constructor. But we want to convey that this WebRidirectCredential requires "interaction" for authorization, for this purpose.

Reply:
For the new credential - we didn't want to keep the name as `AuthorizationCodeCredential` because this name doesn't clearly tell us that it is about OAuth2.0. The name ``AuthorizationCodeCredential` may also tell me (as a user when I hear for the first time) that I am getting some code like a Two factor authentication.

We thought of keeping `WebRedirectCredential`, since it is more intentional about the fact that there is a redirection happening on the web, but we are open to more naming suggestions.

Reply:
IdentityRedirectCredential

Reply:
AuthCodeRedirect